### ISSUE TITLE
VDP R#18 (SET ADJUST) lack of accurate emulation

### ISSUE THREAD

**Issue by nwlg** [openMSX/openMSX#1459]
Hello there.
I've been developing a new game for MSX2 using SET ADJUST (VDP R#18) for scrolling and a split screen for the scoreboard. The thing is it worked great on OpenMSX 18 but it didn't work well on real MSX (tested on TurboR ST & a couple of MSX2/2+).

![image](https://user-images.githubusercontent.com/531709/201097850-2b82c119-886d-4da1-a1be-8e4ac7ad4169.png)

Score area was randomly flickering up and down one pixel on real MSX whilst stayed still on OpenMSX 18.
After reading this topic, https://msx.org/forum/msx-talk/development/showing-horizontal-lines-with-set-adjust-vdp-register
I've fixed the issue waiting for the horizontal retrace (VDP HR bit status) and setting my new SET ADJUST straight away.

It would be great having this behaviour implemented on OpenMSX, basically 'don't write on VDP R#18 whenever you want' in order to provide a more faithfull emulation (and great for developing purposes).

If you need the source code of the game before and after my fix, please contact me anytime (bitvision@gmail.com)

Hope it helps.
Thanks.

**Comment by m9710797** [comment]
In principle I agree with you. Ideally openMSX should reproduce all "bugs/glitches" that will manifest on real MSX machines.

Now specifically about this bug: it's not yet well understood which are the *exact* conditions needed to trigger it, and then what *exactly* should happen. In that case we'd like to take a more conservative approach. So if we're unsure whether some scenario should show a glitch/bug, we'd like to stay on the safe side and make openMSX not show the bug/glitch.

In other words: making an error where openMSX shows a bug but the real MSX machine works fine is worse than the other way around.

If you do find out the *exact* behavior, then please let us know.
Thanks.

**Comment by nwlg** [comment]
ok, understood.
Thanks.

**Comment by MBilderbeek** [comment]
I'd put it like this: without understanding the _exact_ behaviour, a 'fixing' this glitch may cause regressions for other cases which suddently change from 'emulated correctly' to 'showing glitches'. But if the _exact_ behaviour is known, this is much less likely.

**Comment by MBilderbeek** [comment]
Having that said, @nwlg can you post a test binary that shows the issue on real MSX (and not openMSX)? Feel free to strip it a lot, if you don't want to publish a full game for a test here.
